File: src/main/java/com/anod/appwatcher/wishlist/ResultsAdapterWishlist.java
package com.anod.appwatcher.wishlist;

import android.content.Context;

import com.anod.appwatcher.market.WishlistEndpoint;
import com.anod.appwatcher.model.WatchAppList;
import com.anod.appwatcher.search.ResultsAdapter;
import com.google.android.finsky.api.model.Document;

/**
 * @author algavris
 * @date 16/12/2016.
 */

class ResultsAdapterWishlist extends ResultsAdapter {
    private final WishlistEndpoint mEngine;

    ResultsAdapterWishlist(Context context, WishlistEndpoint engine, WatchAppList newAppHandler) {
        super(context, newAppHandler);
        mEngine = engine;
    }

    @Override
    public Document getDocument(int position) {
        boolean isLastPosition = mEngine.getCount() - 1 == position;
        return mEngine.getData().getItem(position, isLastPosition);
    }

    @Override
    public int getItemCount() {
        return mEngine.getCount();
    }

}

File: src/main/java/com/anod/appwatcher/wishlist/ResultsAdapterWishList.kt
package com.anod.appwatcher.wishlist

import android.content.Context

import com.anod.appwatcher.content.WatchAppList
import com.anod.appwatcher.search.ResultsAdapter
import finsky.api.model.Document
import info.anodsplace.playstore.WishlistEndpoint

/**
 * @author Alex Gavrishev
 * *
 * @date 16/12/2016.
 */

internal class ResultsAdapterWishList(context: Context, private val endpoint: WishlistEndpoint, viewModel: WishlistViewModel) : ResultsAdapter(context, viewModel) {

    override fun document(position: Int): Document {
        val isLastPosition = endpoint.count - 1 == position
        return endpoint.listData!!.getItem(position, isLastPosition)!!
    }

    override fun getItemCount(): Int {
        return endpoint.count
    }

}
